app/scripts/views/components/barChart/barChart.js

```diff
@@ -306,27 +306,29 @@
       var hasGreaterOutlier = false;
       var hasSmallerOutlier = false;
       var startNumIndex = 0;
-      var endNumIndex = opts_.xDomain.length - 1;
+      var endNumIndex = opts_.xDomain.length >= 1 ? opts_.xDomain.length - 1 : 0;
```

Comment:
So if `opts_.xDomain` is empty, then endNumIndex is 0, then what's gonna happen at line 359?